sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/OkHttpHttpClient.java

```diff
@@ -138,33 +138,36 @@ private static long getRequestContentLength(BinaryData content, Headers headers)
         return contentLength;
     }
 
-    private static Response<?> toResponse(HttpRequest request, okhttp3.Response response, boolean eagerlyReadResponse,
-                                          boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
-        /*// For now, eagerlyReadResponse and ignoreResponseBody works the same.
-        if (ignoreResponseBody) {
-            ResponseBody body = response.body();
-
-            if (body != null) {
-                if (body.contentLength() > 0) {
-                    LOGGER.log(LogLevel.WARNING, () -> "Received HTTP response body when one wasn't expected. "
-                        + "Response body will be ignored as directed.");
-                }
+    private Response<?> toResponse(HttpRequest request, okhttp3.Response response, boolean eagerlyReadResponse,
+                                   boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
+        okhttp3.Headers responseHeaders = response.headers();
 
-                body.close();
+        if (isTextEventStream(responseHeaders) && response.body() != null) {
+            ServerSentEventListener listener = request.getServerSentEventListener();
+            if (listener != null) {
+                processTextEventStream(request, httpRequestConsumer -> this.send(request), response.body().byteStream(), listener, LOGGER);
+            } else {
+                throw LOGGER.logThrowableAsError(new RuntimeException("No listener attached to the request."));
             }
-
             return new OkHttpResponse(response, request, eagerlyConvertHeaders, EMPTY_BODY);
-        }*/
+        }
+        return processResponse(request, response, eagerlyReadResponse, ignoreResponseBody, eagerlyConvertHeaders);
+    }
 
-        // Use a buffered response when we are eagerly reading the response from the network and the body isn't empty.
+    private Response<?> processResponse(HttpRequest request, okhttp3.Response response, boolean eagerlyReadResponse,
+        boolean ignoreResponseBody, boolean eagerlyConvertHeaders) throws IOException {
         if (eagerlyReadResponse || ignoreResponseBody) {
             try (ResponseBody body = response.body()) {
                 byte[] bytes = (body != null) ? body.bytes() : EMPTY_BODY;
-
                 return new OkHttpResponse(response, request, eagerlyConvertHeaders, bytes);
             }
         } else {
             return new OkHttpResponse(response, request, eagerlyConvertHeaders, null);
         }
     }
+
+    private static boolean isTextEventStream(okhttp3.Headers responseHeaders) {
+        return responseHeaders != null
+            && ContentType.TEXT_EVENT_STREAM.equals(responseHeaders.get(HeaderName.CONTENT_TYPE.toString()));
```

Comment:
`equals` check will not work for all scenarios as services can send the content type as `text/event-stream; charset=utf8` which should still be treated as SSE. So, we should use `startsWith` instead of `equals`.

Reply:
I think in all spots we're checking for the content type we should use

```java
ContentType.TEXT_EVENT_STREAM.regionMatches(true, 0, responseHeaders.get(HeaderName.CONTENT_TYPE.toString()), 0, ContentType.TEXT_EVENT_STREAM.size())
```

This will perform a case-insensitive comparison without instantiating a new string.